src/coreclr/jit/codegenarm64.cpp

```diff
@@ -4209,26 +4209,55 @@ void CodeGen::genCodeForStoreInd(GenTreeStoreInd* tree)
         {
             bool addrIsInReg   = addr->isUsedFromReg();
             bool addrIsAligned = ((tree->gtFlags & GTF_IND_UNALIGNED) == 0);
+            bool handledByRcpc = false;
 
-            if ((ins == INS_strb) && addrIsInReg)
+            // On arm64-v8.4+ we can use stlur* instructions with acquire/release semantics
+            // if the address is LEA with just imm offset (unscaled)
+            if (compiler->compOpportunisticallyDependsOn(InstructionSet_Rcpc2) && addrIsAligned &&
```

Comment:
Maybe you should make this symmetric with the ldapur implementation, and only require `addrIsAligned` for the non-byte cases (INS_strh, INS_str)?

Reply:
thanks, unified code with loads to have a single method to decide which load/store instruction to choose.